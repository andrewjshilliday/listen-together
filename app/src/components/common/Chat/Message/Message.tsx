import React from 'react';
import styled from 'styled-components';

interface MessageProps {
  message: any,
  name: string
}

const Message: React.FC<MessageProps> = ({ message, name }) => {
  let isSentByCurrentUser = false;
  const trimmedName = name.trim().toLowerCase();

  if(message.user === trimmedName) {
    isSentByCurrentUser = true;
  }

  return (
    <MessageContainer isSentByCurrentUser={isSentByCurrentUser}>
      {!isSentByCurrentUser && <Username>{message.user}</Username>}
      <MessageText isSentByCurrentUser={isSentByCurrentUser}>{message.text}</MessageText>
    </MessageContainer>
  );
}

export default Message;


const MessageContainer = styled.div<{isSentByCurrentUser: boolean}>`
  display: flex;
  flex-direction: column;
  flex-shrink: 0;
  max-width: 75%;
  min-width: 50%;
  overflow: hidden;
  ${props => props.isSentByCurrentUser && `
    align-self: flex-end;
  `}
`;
const Username = styled.div`
  margin: 0 15px 0 30px;
  font-size: 10pt;
  color: gray;
`;
const MessageText = styled.div<{isSentByCurrentUser: boolean}>`
  padding: 10px;
  color: white;
  ${props => props.isSentByCurrentUser && `
    border-radius: 2em 2em 0 2em;
    margin: 3px 10px;
    background: blue;
  `}
  ${props => !props.isSentByCurrentUser && `
    border-radius: 2em 2em 2em 0;
    margin: 3px 10px 3px 10px;
    background: green;
  `}
`;
