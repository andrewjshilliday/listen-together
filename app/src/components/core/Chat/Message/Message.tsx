import React from 'react';
import { useChat } from '../../../providers';
import { TextField } from '@material-ui/core';
import styled from 'styled-components';

interface MessageProps {
  message: any,
  name: string
}

const Message: React.FC<MessageProps> = ({ message, name }) => {
  let isSentByCurrentUser = false;
  const trimmedName = name.trim().toLowerCase();

  if(message.user === trimmedName) {
    isSentByCurrentUser = true;
  }

  return (
    <MessageContainer isSentByCurrentUser={isSentByCurrentUser}>
      {!isSentByCurrentUser && <Username>{name}</Username>}
      <MessageText isSentByCurrentUser={isSentByCurrentUser}>{message.text}</MessageText>
    </MessageContainer>
  );
}

export default Message;


const MessageContainer = styled.div<{isSentByCurrentUser: boolean}>`
  display: flex;
  flex-direction: column;
  max-width: 75%;
  min-width: 50%;
  ${props => props.isSentByCurrentUser && `
    align-self: flex-end;
  `}
`;
const Username = styled.div`
  margin: 5px 15px 0 30px;
  font-size: 10pt;
  color: gray;
`;
const MessageText = styled.div<{isSentByCurrentUser: boolean}>`
  padding: 10px;
  color: white;
  border-radius: 2em;
  ${props => props.isSentByCurrentUser && `
    margin: 10px;
    background: blue;
  `}
  ${props => !props.isSentByCurrentUser && `
    margin: 3px 10px 10px 10px;
    background: green;
  `}
`;
