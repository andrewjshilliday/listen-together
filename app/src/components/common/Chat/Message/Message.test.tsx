import React from 'react';
import renderer from 'react-test-renderer';
import Message from './Message';

it('renders correctly', () => {
  const tree = renderer.create(<Message message='test' name='test' />).toJSON();
  expect(tree).toMatchSnapshot();
});
