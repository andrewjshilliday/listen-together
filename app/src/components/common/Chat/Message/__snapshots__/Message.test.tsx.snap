// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm bISjJK"
>
  <div
    className="sc-AxirZ iWqrgE"
  />
  <div
    className="sc-AxiKw jyHTFv"
  />
</div>
`;
